stl/inc/atomic

```diff
@@ -115,8 +115,23 @@ _NODISCARD extern "C" bool __cdecl __std_atomic_has_cmpxchg16b() noexcept;
 #define ATOMIC_LLONG_LOCK_FREE    2
 #define ATOMIC_POINTER_LOCK_FREE  2
 
+// Padding bits should not participate in cmpxchg comparison starting in C++20.
+// Clang does not have __builtin_zero_non_value_bits to exclude these bits to implement this C++20 feature.
+// The EDG front-end substitutes everything and runs into incomplete types passed to atomic<T>.
+#if _HAS_CXX20 && !defined(__clang__) /* TRANSITION, LLVM-46685 */ && !defined(__EDG__)

```

Comment:
Hmm should we do this always even when not C++20? It seems a straight up correctness win in all modes.

Reply:
I'm also asking https://github.com/microsoft/STL/pull/1029#issuecomment-657060940 :-)

I agree on one hand, on the other, remember atomic constructor?

Reply:
The atomic constructor situation is different because it had "static initialization order fiasco" effects not present here.

Reply:
Sure, but I can imagine other ways, like defining a type with `=` as memcpy` and `==` as `memcmp`, thus effectively making its padding bits participate in value. I think old Standard explicitly specifies bitwise CAS instead of leaving out this case undefined.

(Sure that's not a normal use case problem, but neither is this whole feature)

Reply:
(if the decision is made towards exposing in pre-C++20, be sure to replace `constexpr` with `_CONSTEXPR_IF`)

Reply:
>(Sure that's not a normal use case problem, but neither is this whole feature)

Indeed.

I'm OK with leaving it '20 only.